oncrpc4j-core/src/main/java/org/dcache/xdr/RpcCall.java

```diff
@@ -341,16 +352,18 @@ public void call(int procedure, XdrAble args, CompletionHandler<RpcReply, XdrTra
      * @param procedure The number of the procedure.
      * @param args The argument of the procedure.
      * @param type The expected type of the reply
+     * @param timeoutValue timeout value. 0 means no timeout
+     * @param timeoutUnits units for timeout value
      * @return A Future representing the result of the operation.
      * @throws OncRpcException
      * @throws IOException
      * @since 2.4.0
      */
-    public <T extends XdrAble> Future<T> call(int procedure, XdrAble args, final Class<T> type)
-            throws OncRpcException, IOException {
+    public <T extends XdrAble> Future<T> call(int procedure, XdrAble args, final Class<T> type, long timeoutValue, TimeUnit timeoutUnits)
```

Comment:
I think there are two use cases:  code actively calls Future#get, or you register a ballback. In the second case, you want to be notified only with in a time window. I don't see a reason to have a Future and timeout at the same time.